doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1257,237 +1257,211 @@ Managing the local context
+   .. prodn::
+      alias_definition ::= ( @ident {* @simple_binder } := @term )
+      simple_binder ::= @name
+      | ( {+ @name } : @term )
+      as_name ::= as @ident
 
-      This notation allows specifying which occurrences of :token:`term` have
-      to be substituted in the context. The :n:`in @goal_occurrences` clause
-      is an occurrence clause whose syntax and behavior are described in
-      :ref:`goal occurrences <occurrenceclauses>`.
+   The first form adds a new local definition
+   :n:`@ident := fun {* @simple_binder } => @term` and replaces :n:`@term` in
+   the goal with a new variable :n:`@ident`.
 
-   .. tacv:: set (@ident {* @binder } := @term) {? in @goal_occurrences }
+   .. todo PR above: should be replaces term with @ident followed by the binding values
+      Perhaps you could provide a simple example?
 
-      This is equivalent to :n:`set (@ident := fun {* @binder } => @term) {? in @goal_occurrences }`.
+   The second form is equivalent to :n:`set (@ident := @one_term) {? @occurrences }`
+   using :n:`@ident`, if present, or auto-generated if not provided.
 
-   .. tacv:: set @term {? in @goal_occurrences }
+   If :token:`term` or :token:`one_term` has holes (i.e. subexpressions with the
+   form “`_`”), the tactic first checks that all subterms matching the pattern
+   are compatible before doing the replacement using the leftmost subterm
+   matching the pattern.
 
-      This behaves as :n:`set (@ident := @term) {? in @goal_occurrences }`
-      but :token:`ident` is generated by Coq.
+   .. exn:: The variable @ident is already declared.
+      :undocumented:
 
-   .. tacv:: eset (@ident {* @binder } := @term) {? in @goal_occurrences }
-             eset @term {? in @goal_occurrences }
+   .. tacn:: eset @alias_definition {? @occurrences }
+             eset @one_term {? @as_name } {? @occurrences }
       :name: eset; _
 
-      While the different variants of :tacn:`set` expect that no existential
-      variables are generated by the tactic, :tacn:`eset` removes this
-      constraint. In practice, this is relevant only when :tacn:`eset` is
+      Similar to :tacn:`set`, but generates an existential variable when
+      a variable can't be instantiated instead of failing.
+      In practice, this is relevant only when :tacn:`eset` is
       used as a synonym of :tacn:`epose`, i.e. when the :token:`term` does
       not occur in the goal.
 
-.. tacn:: remember @term as @ident__1 {? eqn:@naming_intropattern }
-   :name: remember
+.. tacn:: remember @one_term {? @as_name } {? eqn : @naming_intropattern } {? in @goal_occurrences }
 
-   This behaves as :n:`set (@ident := @term) in *`, using a logical
-   (Leibniz’s) equality instead of a local definition.
+   Equivalent to :n:`set (@ident := @term) in *` but uses :term:`Leibniz equality`
+   instead of a :term:`local definition <context-local definition>`.
    Use :n:`@naming_intropattern` to name or split up the new equation.
```

Comment:
```suggestion
   Use :n:`@naming_intropattern` to name the new equation.
```

Since it's just a naming intropattern, it can't do more than naming...